plugins/network-elements/nuage-vsp/src/com/cloud/network/guru/NuageVspGuestNetworkGuru.java

```diff
@@ -339,34 +398,63 @@ private void updateDhcpOptionsForExistingVms(Network network, HostVO nuageVspHos
         }
     }
 
+    private void checkMultipleSubnetsCombinedWithUseData(Network network){
+        if (_ntwkOfferingSrvcDao.listServicesForNetworkOffering(network.getNetworkOfferingId()).contains(Network.Service.UserData.getName()) && _vlanDao.listVlansByNetworkId(network.getId()).size() > 1) {
+            List<VlanVO> vlanVOs = _vlanDao.listVlansByNetworkId(network.getId());
+            for(VlanVO vlanVoItem : vlanVOs){
+                for(VlanVO VlanVoItem2 : vlanVOs){
+                    if(!vlanVoItem.equals(VlanVoItem2) && !VlanVoItem2.getVlanGateway().equals(vlanVoItem.getVlanGateway())){
+                        s_logger.error("NuageVsp provider does not support multiple subnets in combination with user data.");
```

Comment:
Please add identifying information of the duplicated vlan/subnet and network to the error message to assist with forensic debugging.